overlord/configstate/config/helpers.go

```diff
@@ -35,14 +35,21 @@ import (
 
 var validKey = regexp.MustCompile("^(?:[a-z0-9]+-?)*[a-z](?:-?[a-z0-9])*$")
 
+func ValidateKey(key string) error {
```

Comment:
Please document the function.

Reply:
I think the function also needs unit tests for various valid and invalid cases. This would complement the changes in `cmd/snap/cmd_set_test.go`, but the bulk of the testing should be related to this function in particular, as it's exported.

Reply:
@zyga Documentation done!

@olivercalder Agree! I've written tests for the `ValidateKey` and `ParseKey` functions.